docs/schema/relations.md

```diff
@@ -0,0 +1,295 @@
+}
+
+export class Tag {
+	posts = def.oneHasMany(PostTag, 'tag')
+}
+```
+
+### One-has-one
+
+There is (at most) one entity on each side of this relationship.
+
+![one has one relation](/assets/one-has-one.svg)
+
+- We define owning side of this relation using `oneHasOne` method. 
+- Optionally, we define an inverse side using `oneHasOneInverse` method. 
+- Joining column with actual relationship value is located on owning side.
+- For this relationship, you can also configure 
+  - [nullability](#relationships-nullability) on both sides
+  - [delete behaviour](#on-delete-behavior) on owning side 
+  - [orphan removal](#orphan-removal) on owning side
+
+#### Use case
+
+Not as common, but sometimes useful type of relationship. 
+Imagine entities *Post* and *PostContent* - there is always single *PostContent* entity of each *Post* and a single *Post* for each *PostContent*. 
+In this case, it might seem a bit pointless - all fields *PostContent* entity can be safely inlined into *Post*. 
+Let's change it a bit - rename *PostContent* to *Content*. 
+Now we can reference this generic *Content* not only from a *Post*, but also from e.g. a *Category* and use same logic for storing, managing and rendering the *Content* of both entities. 
+In this example, owning side would be in *Post* and *Category* entities, optional inverse side in *Content*.
+
+#### Example: Configuring only owning side
+
+```typescript
+export class Post {
+	// highlight-next-line
+	content = def.oneHasOne(Content)
+}
+export class Content {
+}
+```
+
+#### Example: Configuring both owning and inverse side
+
+```typescript
+export class Post {
+	// highlight-next-line
+	content = def.oneHasOne(Content, 'post')
+}
+
+export class Content {
+	// highlight-next-line
+	post = def.oneHasOneInverse(Post, 'content')
+}
+```
+
+## Relationships settings
+
+### Nullability
+
+You can also define `.notNull()` constraint for "one has one" relationships and owning side of "many has one" relationship.
+
+#### Example: making category of post not nullable
+```typescript
+export class Post {
+	// highlight-next-line
+	category = def.manyHasOne(Category).notNull();
+}
+```
+
+### On delete behavior
+
+Using `.onDelete()` you can set what happens when referenced entity is deleted. 
+E.g. you have a post, which is assigned to a category. When a category is deleted, three things can happen:
+
+- restrict: this is default behavior. When you try to delete an entity, which is referenced from other entities, the delete operation will fail.
+- set null: field, which references removed entity, is set to null. Obviously, this is possible only for nullable relationships. You can use shortcut `.setNullOnDelete()` to select this behavior.
+- cascade: all entities, which references an entity which is being removed, are also removed. You can use a shortcut `.cascadeOnDelete()`.
+
+Pay attention when you are choosing the strategy, because choosing a wrong strategy may lead to runtime errors.
```

Comment:
```suggestion
Pay attention when you are choosing the strategy, because choosing a wrong strategy may lead to runtime errors or deleting more content than you wanted.
```